### PR TITLE
[SWCR] Add new object type Software Component Relations

### PR DESCRIPTION
**Pull request by hottek**

### CODE REVIEW

**Review by schneidermic0 — CHANGES_REQUESTED** [review]
Thanks @hottek. Please find my comments/questions below.

File: file-formats/swcr/type/zif_aff_swcr_v1.intf.abap [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    BEGIN OF ty_relation,
+      "! <p class="shorttext">Related Software Component</p>
+      "! Related Software Component
+      "! $required
+      related_software_component TYPE c LENGTH 30,
+      "! <p class="shorttext">Relation Type</p>
+      "! Relation Type
+      "! $required
+      relation_type              TYPE ty_relation_type,
+    END OF ty_relation.
```
[the remark this answers is not in the document]
Comment (hottek) [reply file-formats/swcr/type/zif_aff_swcr_v1.intf.abap:33]: I agree, will change it to `software_component` and `type`.

File: file-formats/swcr/type/zif_aff_swcr_v1.intf.abap [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    "! Software Component Relations
+    BEGIN OF ty_main,
+      "! <p class="shorttext">Format Version</p>
+      "! Format version
+      "! $required
+      format_version TYPE zif_aff_types_v1=>ty_format_version,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header         TYPE zif_aff_types_v1=>ty_header_60_cloud,
```
[the remark this answers is not in the document]
Comment (hottek) [reply file-formats/swcr/type/zif_aff_swcr_v1.intf.abap:50]: That would be great.

[the remark this answers is not in the document]
Comment (schneidermic0) [reply file-formats/swcr/type/zif_aff_swcr_v1.intf.abap]: I have just created an own header for the object. I want to see whether we need this for more AFF types before we add it to `zif_aff_types_v1`.

See https://github.com/SAP/abap-file-formats/pull/539/commits/ec5ee0d3c6d10f2e1444c5d259d1bb1a712364bf

**Comment by hottek** [comment]
@schneidermic0 Thank you, the changes look good to me.

**Review by schneidermic0 — APPROVED** [review]
Thanks, @hottek 

Looks good to me.

File: file-formats/swcr/type/zif_aff_swcr_v1.intf.abap [hunk 3]
```diff
@@ -10,57 +10,52 @@ INTERFACE zif_aff_swcr_v1
...
       "! Header
       "! $required
       header         TYPE ty_header,
-      "! <p class="shorttext">Relations</p>
-      "! Relations
-      "! $required
-      relations      TYPE ty_relations,
+      "! <p class="shorttext">Access Permissions</p>
+      "! The Software Component Grants Access Permission to the Specified Software Components
+      permissions    TYPE ty_permissions,
```
Comment (schneidermic0) [inline file-formats/swcr/type/zif_aff_swcr_v1.intf.abap:55]: I am just wondering, whether we should keep the short text and the attribute name in sync. I guess you know better whether "access permissions" or just "permissions" is the better term.

File: file-formats/swcr/type/zif_aff_swcr_v1.intf.abap [hunk 4]
```diff
@@ -10,57 +10,52 @@ INTERFACE zif_aff_swcr_v1
...
-  "! $values {@link zif_aff_swcr_v1.data:co_relation_type}
-  TYPES ty_relation_type TYPE c LENGTH 1.
+  TYPES:
+    "! <p class="shorttext">Access Permission</p>
+    "! Access Permission
+    BEGIN OF ty_permission,
+      "! <p class="shorttext">Software Component</p>
+      "! Software Component
+      "! $required
+      software_component TYPE c LENGTH 30,
```
Comment (schneidermic0) [inline file-formats/swcr/type/zif_aff_swcr_v1.intf.abap:20]: Just a question: The permission type. Previously relation type isn't needed any more?

**Review by schneidermic0 — CHANGES_REQUESTED** [review]
Thanks, @hottek. 

I found only some minor issues

File: file-formats/swcr/type/zif_aff_swcr_v1.intf.abap (hunk 3 above)
Comment (hottek) [reply file-formats/swcr/type/zif_aff_swcr_v1.intf.abap:55]: We would like to keep the "permissions" term for the attribute.

File: file-formats/swcr/type/zif_aff_swcr_v1.intf.abap (hunk 4 above)
Comment (hottek) [reply file-formats/swcr/type/zif_aff_swcr_v1.intf.abap:20]: That is correct. We now implicitly derive the permission type from the software component.

File: file-formats/swcr/type/zif_aff_swcr_v1.intf.abap (hunk 3 above)
Comment (schneidermic0) [reply file-formats/swcr/type/zif_aff_swcr_v1.intf.abap:55]: One more Would you then also suggest to name the shorttext just "Permission"? Or is there any reason why you don't want to keep the shorttext and the field name in sync?

Comment (hottek) [reply file-formats/swcr/type/zif_aff_swcr_v1.intf.abap:55]: For the time being, all supported types of permissions are access permissions. Therefore we want to keep the shorttext as specific and understandable as possible, i.e. we would like to stay with "Access Permissions". As soon as there are use cases for other types of permissions we would adjust the shorttext accordingly.

Comment (schneidermic0) [reply file-formats/swcr/type/zif_aff_swcr_v1.intf.abap:55]: OK, I understand. Thanks for the clarification

**Review by schneidermic0 — APPROVED** [review]
Looks good to me

**Review by wurzka — CHANGES_REQUESTED** [review]
We performed an UX Cross Check. I only found one issue regarding the descriptions. See my comments below.

**Review by wurzka — APPROVED** [review]
LGTM :)

**Pull request merged**